build.gradle

```diff
@@ -84,6 +84,25 @@ allprojects {
 
     project.getExtensions().getExtraProperties().set("versions", VersionProperties.getVersions());
 
+    configurations {
+        agent
+    }
+
+    dependencies {
+        agent "org.opensearch:opensearch-agent-bootstrap:${opensearch_version}"
+        agent "org.opensearch:opensearch-agent:${opensearch_version}"
+        agent "net.bytebuddy:byte-buddy:1.17.5"
+    }
+
+    tasks.withType(Test) {
+        dependsOn prepareAgent
+        jvmArgs += ["-javaagent:" + project.layout.buildDirectory.file("agent/opensearch-agent-${opensearch_version}.jar").get()]
+    }
+
+    task prepareAgent(type: Copy) {
+       from(configurations.agent)
+       into "$buildDir/agent"
+    }
```

Comment:
I think this should not be nested within `allprojects`?

Reply:
Yeah, i also think so. figuring that out.